File: solutions/dart/resistor-color-trio/1/lib/resistor_color_trio.dart
import 'dart:math';

class ResistorColorTrio {
  static const Map<String, int> colorValues = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9,
  };

  String label(List<String> colors) {
    int first = colorValues[colors[0]]!;
    int second = colorValues[colors[1]]!;
    int multiplier = colorValues[colors[2]]!;
    int ohms = (first * 10 + second) * pow(10, multiplier).toInt();

    if (ohms < 1000) {
      return '$ohms ohms';
    } else if (ohms < 1000000) {
      int kilo = ohms ~/ 1000;
      return '$kilo kiloohms';
    } else if (ohms < 1000000000) {
      int mega = ohms ~/ 1000000;
      return '$mega megaohms';
    } else {
      int giga = ohms ~/ 1000000000;
      return '$giga gigaohms';
    }
  }
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
def label(colors):
    colors_list=[
        'black', 'brown', 'red',
        'orange', 'yellow', 'green',
        'blue', 'violet', 'grey',
        'white'
    ]

    digits=colors[0:3]
    value1=digits[0]
    value2=digits[1]
    suffix=digits[2]

    value=int(str(colors_list.index(str(value1)))+str(colors_list.index(str(value2))))
    multiplier=pow(10, colors_list.index(str(suffix)))
    final_value=value*multiplier

    if 0<=final_value<1000:
        return str(final_value)+' ohms'
    
    if 1000<=final_value<1000000:
        return str(int(final_value/1000))+' kiloohms'

    if 1000000<=final_value<1000000000:
        return str(int(final_value/1000000))+' megaohms'
    
    return str(int(final_value/1000000000))+' gigaohms'



